tests/storages_tests/test_storages.py

```diff
@@ -1060,3 +1060,34 @@ def test_get_best_trial_for_multi_objective_optimization(storage_mode: str) -> N
             storage.create_new_trial(study_id, template_trial=template_trial)
         with pytest.raises(ValueError):
             storage.get_best_trial(study_id)
+
+
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_trial_id_from_study_id_trial_number(storage_mode: str) -> None:
+
+    with StorageSupplier(storage_mode) as storage:
+        with pytest.raises(KeyError):  # Matching study does not exist.
+            storage.get_trial_id_from_study_id_trial_number(study_id=0, trial_number=0)
+
+        study_id = storage.create_new_study()
+
+        with pytest.raises(KeyError):  # Matching trial does not exist.
+            storage.get_trial_id_from_study_id_trial_number(study_id, trial_number=0)
+
+        trial_id = storage.create_new_trial(study_id)
+
+        assert trial_id == storage.get_trial_id_from_study_id_trial_number(
+            study_id, trial_number=0
+        )
+
+        # Trial IDs are globally unique within a storage but numbers are only unique within a
+        # study. Create a second study within the same storage.
+        study_id = storage.create_new_study()
+
+        storage.create_new_trial(study_id)
+        storage.create_new_trial(study_id)
+        trial_id = storage.create_new_trial(study_id)
+
+        assert trial_id == storage.get_trial_id_from_study_id_trial_number(
+            study_id, trial_number=2
+        )
```

Comment:
I feel it is unclear what this test checks.

Reply:
It's testing the behavior of `get_trial_id_from_study_id_trial_number` when a storage contains more than one study. I.e., where the trial ID and the trial numbers are offset. I tried to explain it in the comments on line 1083, 1084. Can we improve this test function?

Reply:
Simplified the test to avoid confusion. You're right it quite meaningless.